amplipi/updater/asgi.py

```diff
@@ -95,6 +98,80 @@ class ReleaseInfo(BaseModel):
   logger.exception(f'Error loading identity file: {e}')
 
 
+@router.get("/settings/persist_logs")
+def get_log_persist_state():
+  """
+  Checks /etc/systemd/journald.conf to find if the current storage setting is persistent and returns a bool
+  Note that returning false doesn't necessarily mean that logs are set to volatile, and could just mean that the config file is missing the line being read
+  """
+  conf_path = '/etc/systemd'
+  config = configparser.ConfigParser(strict=False, allow_no_value=True)
+  config.read(f'{conf_path}/journald.conf')
+  return config.get("Journal", "Storage", fallback="") == "persistent"
+
+
+@router.post("/settings/persist_logs")
+def toggle_persist_logs(value: Optional[bool] = None):
+  """
+  Toggles the option within journald to save logs to memory or storage.
+  Accepts an option bool called "value" for if you wish to change the value to a specific option rather than simply flip the state of the config file
+  """
+  try:
+    conf_path = '/etc/systemd'
+    config = configparser.ConfigParser(strict=False, allow_no_value=True)
+    config.read(f'{conf_path}/journald.conf')
+
+    if 'Journal' not in config:
+      config.add_section('Journal')
+
+    # goal_value is a holdover from when this setting was in app.py and toggled via the config settings page
+    # the bool is true if you wish to turn persistent logging on and false if you wish to turn it off
+    # It can be set either with the optional arg of the function or it attempts to find the current value from the config file and set the opposite of that
+    goal_value: bool
+    if value is not None:
+      goal_value = value
+    else:
+      goal_value = not get_log_persist_state()  # goal_value is opposite of present value as a default/fallback
+
+    if goal_value:  # Set persist
+      config.set('Journal', 'Storage', 'persistent')
+      config.set('Journal', 'SyncIntervalSec', '30s')
+      config.set('Journal', 'SystemMaxUse', '64M')
+
+      config.remove_option('Journal', 'RuntimeMaxUse')
+      config.remove_option('Journal', 'ForwardToConsole')
+      config.remove_option('Journal', 'ForwardToWall')
+
+      logger.info("persist_logs activated!")
+
+    else:  # Reset config to default as seen in configure.py
+      config.set('Journal', 'Storage', 'volatile')
+      config.set('Journal', 'RuntimeMaxUse', '64M')
+      config.set('Journal', 'ForwardToConsole', 'no')
+      config.set('Journal', 'ForwardToWall', 'no')
+
+      config.remove_option('Journal', 'SyncIntervalSec')
+      config.remove_option('Journal', 'SystemMaxUse')
```

Comment:
this strategy is somewhat brittle; what happens when an update is installed?

Reply:
Haven't had any updates that contain this code to test with, I imagine it just resets back to volatile as is defined in `configure.py`

Reply:
you're correct; I would consider that a bug with this implementation

Reply:
We initialize Amplipi's with an image right?
If so, would an acceptable solution be simply ripping that chunk of `configure.py` out?

I struggle to think of a method where I could check if persistent logs were set on purpose vs being set because that's the system default prior to the first `configure.py` run

Reply:
Alternative: https://github.com/micro-nova/AmpliPi/pull/828/files#r1707793661

Reply:
your "alternative" link is broken.

we create the image using `configure.py`, so directly ripping that chunk out will not work. i would consider storing this state somewhere in the config dir, and using that as inputs to both configure.py and `get_log_perist_state` to set the correct values, with an uninitialized default of `volatile`

Reply:
That link was meant to direct to this line in configure.py that only creates the file if it doesn't already exist:
`r'[ ! -d /var/log/journal ] && echo -e "[Journal]\nStorage=volatile\nRuntimeMaxUse=64M\nForwardToConsole=no\nForwardToWall=no\n" | sudo tee /etc/systemd/journald.conf',`